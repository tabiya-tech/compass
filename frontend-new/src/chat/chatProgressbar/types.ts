export enum ConversationPhase {
  INITIALIZING = "INITIALIZING",
  INTRO = "INTRO",
  COLLECT_EXPERIENCES = "COLLECT_EXPERIENCES",
  DIVE_IN = "DIVE_IN",
  ENDED = "ENDED",
  UNKNOWN = "UNKNOWN"
}

export type CurrentPhase = {
  percentage: number
  phase: ConversationPhase
}

export const defaultCurrentPhase = {
  phase: ConversationPhase.INITIALIZING,
  percentage: 0
}
