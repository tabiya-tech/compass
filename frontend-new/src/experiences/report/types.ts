import { Experience } from "src/experiences/experiencesDrawer/experienceService/experiences.types";

export type ReportProps = {
  name: string;
  email: string;
  phone: string;
  address: string;
  experiences: Experience[];
  conversationConductedAt: string | null;
};

export enum ExportFormat {
  PDF = "pdf",
  DOCX = "docx",
}

export interface IReportFormatProvider {
  download: () => void;
}
