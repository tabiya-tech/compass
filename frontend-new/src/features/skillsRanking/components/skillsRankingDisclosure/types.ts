export const jobSeekerComparisonLabels = ["LOWEST", "SECOND LOWEST", "MIDDLE", "SECOND HIGHEST", "HIGHEST"]