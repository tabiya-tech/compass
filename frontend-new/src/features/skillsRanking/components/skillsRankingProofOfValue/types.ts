export enum EffortType {
  TIME_BASED = "time_based",
  WORK_BASED = "work_based",
}