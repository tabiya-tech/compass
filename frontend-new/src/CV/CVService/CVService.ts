import { getBackendUrl } from "src/envService";
import { getRestAPIErrorFactory } from "src/error/restAPIError/RestAPIError";
import { StatusCodes } from "http-status-codes";
import ErrorConstants from "src/error/restAPIError/RestAPIError.constants";
import { customFetch } from "src/utils/customFetch/customFetch";

export default class CVService {
  private static instance: CVService;
  readonly cvEndpointUrl: string;
  readonly apiServerUrl: string;

  private constructor() {
    this.apiServerUrl = getBackendUrl();
    this.cvEndpointUrl = `${this.apiServerUrl}/users`;
  }

  /**
   * Get the singleton instance of the CVService.
   * @returns {CVService} The singleton instance of the CVService.
   */
  static getInstance(): CVService {
    if (!CVService.instance) {
      CVService.instance = new CVService();
    }
    return CVService.instance;
  }

  public async uploadCV(userId: string, file: File): Promise<string[]> {
    const serviceName = "CVService";
    const serviceFunction = "uploadCV";
    const method = "POST";
    const errorFactory = getRestAPIErrorFactory(serviceName, serviceFunction, method, this.cvEndpointUrl);
    const constructedUploadUrl = `${this.cvEndpointUrl}/${userId}/cv`;

    const resolveContentType = (): string => {
      if (file.type) return file.type;
      const lower = file.name.toLowerCase();
      if (lower.endsWith(".pdf")) return "application/pdf";
      if (lower.endsWith(".docx")) return "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
      if (lower.endsWith(".txt")) return "text/plain";
      return "application/octet-stream";
    };

    const headers: Record<string, string> = {
      Accept: "application/json",
      "Content-Type": resolveContentType(),
      // Backend reads either 'filename' or 'x-filename'. Using custom header triggers preflight, which is OK.
      "x-filename": file.name,
    };

    const response = await customFetch(constructedUploadUrl, {
      method: method,
      body: file, // stream raw file bytes; backend expects raw body, not multipart
      headers,
      expectedStatusCode: StatusCodes.OK,
      serviceName,
      serviceFunction,
      failureMessage: `Failed to upload CV for user ${userId}`,
      expectedContentType: "application/json",
      compressRequestBody: false, // Disable compression for file uploads
    });

    const responseText = await response.text();

    let parseResponse: { experiences_data?: string[] };
    try {
      parseResponse = JSON.parse(responseText);
    } catch (e: any) {
      throw errorFactory(
        response.status,
        ErrorConstants.ErrorCodes.INVALID_RESPONSE_BODY,
        "Response did not contain valid JSON",
        {
          responseText,
          error: e,
        }
      );
    }

    const experiences = Array.isArray(parseResponse.experiences_data) ? parseResponse.experiences_data : [];
    return experiences;
  }
}
