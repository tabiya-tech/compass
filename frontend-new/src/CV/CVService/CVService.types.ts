import { UploadProcessState } from "src/chat/Chat.types";

export type CVListItem = {
  upload_id: string;
  filename: string;
  uploaded_at: string;
  upload_process_state: UploadProcessState;
  experiences_data: string[] | null;
};
