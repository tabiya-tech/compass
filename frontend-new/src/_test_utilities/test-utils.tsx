// Based on https://testing-library.com/docs/react-testing-library/setup/
import React, { ReactElement, ReactNode } from "react";
import { render, renderHook, RenderHookOptions, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";

// Import the Firebase mock utilities
import "src/_test_utilities/firebaseMock";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/isOnlineProvider/IsOnlineProvider";
jest.mock("firebase/compat/app", () => {
  return {
    initializeApp: jest.fn(),
    auth: jest.fn().mockReturnValue({
      signOut: jest.fn(),
      currentUser: {
        getIdToken: jest.fn(),
        getTokenResult: jest.fn(),
      },
      onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
      signInWithPopup: jest.fn(),
      GoogleAuthProvider: { PROVIDER_ID: "google.com" },
    }),
  };
});

jest.mock("firebaseui", () => {
  return {
    auth: {
      AuthUI: {
        start: jest.fn(),
        getInstance: jest.fn().mockReturnValue({
          start: jest.fn(),
          reset: jest.fn(),
        }),
        reset: jest.fn(),
      },
    },
  };
});

jest.mock("@sentry/react", () => ({
  withSentry: (Component: any) => Component,
  getFeedback: jest.fn(),
  withProfiler: (Component: any) => Component,
  wrapCreateBrowserRouter: (Component: any) => Component,
}));

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

function customRenderHook<T>(
  hook: () => T,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { wrapper?: React.ComponentType }
) {
  const CombinedWrapper = ({ children }: { children: ReactNode }) => {
    const Wrapper = options?.wrapper ?? React.Fragment;
    return (
      <AllTheProviders>
        <Wrapper>{children}</Wrapper>
      </AllTheProviders>
    );
  };

  return renderHook(hook, { wrapper: CombinedWrapper, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
export { customRenderHook as renderHook };
