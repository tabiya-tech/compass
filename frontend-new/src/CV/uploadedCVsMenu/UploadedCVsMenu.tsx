import React from "react";
import { Box, Typography, Skeleton, useTheme } from "@mui/material";
import ArrowBackIcon from "@mui/icons-material/ArrowBack";
import InfoIcon from "@mui/icons-material/Info";
import DescriptionOutlinedIcon from "@mui/icons-material/DescriptionOutlined";
import HelpTip from "src/theme/HelpTip/HelpTip";
import PrimaryIconButton from "src/theme/PrimaryIconButton/PrimaryIconButton";
import { CVListItem } from "src/CV/CVService/CVService.types";
import { ConversationPhase } from "src/chat/chatProgressbar/types";
import { getDurationFromNow } from "src/chat/chatMessage/components/chatMessageFooter/components/timestamp/getDurationFromNow/getDurationFromNow";

const uniqueId = "e9b40dcc-8068-45ec-a96a-9feb823f1696";

export const DATA_TEST_ID = {
  UPLOADED_CVS_MENU_CONTENT: `uploaded-cvs-menu-content-${uniqueId}`,
  UPLOADED_CVS_MENU_ARROW_BACK_ICON: `uploaded-cvs-menu-arrow-back-icon-${uniqueId}`,
  UPLOADED_CVS_MENU_HELP_TIP: `uploaded-cvs-menu-help-tip-${uniqueId}`,
  UPLOADED_CVS_MENU_UPLOADED_TEXT: `uploaded-cvs-menu-uploaded-text-${uniqueId}`,
  UPLOADED_CVS_MENU_DESCRIPTION_ICON: `uploaded-cvs-menu-description-icon-${uniqueId}`,
  UPLOADED_CVS_MENU_FILE_NAME: `uploaded-cvs-menu-file-name-${uniqueId}`,
  UPLOADED_CVS_MENU_UPLOAD_DATE: `uploaded-cvs-menu-upload-date-${uniqueId}`,
  UPLOADED_CVS_MENU_SKELETON: `uploaded-cvs-menu-skeleton-${uniqueId}`,
};

interface UploadedCVsMenuContentProps {
  uploadedCVs: CVListItem[];
  onSelect: (cv: CVListItem) => void;
  onBack: () => void;
  isLoading: boolean;
  currentPhase?: ConversationPhase;
}

const UploadedCVsMenu: React.FC<UploadedCVsMenuContentProps> = ({
  currentPhase,
  onBack,
  onSelect,
  uploadedCVs,
  isLoading,
}) => {
  const theme = useTheme();
  const isCollectPhase = currentPhase === ConversationPhase.COLLECT_EXPERIENCES;

  const helpTipText = isCollectPhase
    ? "Tap a CV to load its content into the text field. Review and send when you're ready."
    : "CV selection is only available during the experience collection phase.";

  return (
    <Box
      sx={{ display: "flex", flexDirection: "column", width: "100%", gap: theme.fixedSpacing(theme.tabiyaSpacing.md) }}
      data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_CONTENT}
    >
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          gap: theme.fixedSpacing(theme.tabiyaSpacing.xs),
          px: theme.fixedSpacing(theme.tabiyaSpacing.md),
        }}
      >
        <PrimaryIconButton
          title="Back to main menu"
          onClick={(e) => {
            e.preventDefault();
            e.stopPropagation();
            onBack();
          }}
          sx={{ color: theme.palette.text.secondary }}
          data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_ARROW_BACK_ICON}
        >
          <ArrowBackIcon />
        </PrimaryIconButton>
        <Typography
          variant="subtitle2"
          fontWeight="bold"
          color="text.primary"
          data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_UPLOADED_TEXT}
        >
          {`Previously uploaded CVs (${uploadedCVs.length})`}
        </Typography>
        <HelpTip icon={<InfoIcon />} data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_HELP_TIP}>
          {helpTipText}
        </HelpTip>
      </Box>
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          rowGap: theme.fixedSpacing(theme.tabiyaSpacing.sm),
          maxHeight: theme.fixedSpacing(theme.tabiyaSpacing.xl * 6),
          overflow: "auto",
          px: theme.fixedSpacing(theme.tabiyaSpacing.md),
        }}
      >
        {isLoading && (
          <Box sx={{ display: "flex", flexDirection: "column" }} data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_SKELETON}>
            {Array.from({ length: 2 }).map((_, idx) => (
              <Box
                key={idx}
                sx={{
                  display: "flex",
                  alignItems: "center",
                  py: theme.fixedSpacing(theme.tabiyaSpacing.xs),
                  gap: theme.fixedSpacing(theme.tabiyaSpacing.sm),
                }}
              >
                <Skeleton variant="circular" width={24} height={24} />
                <Box sx={{ flexGrow: 1 }}>
                  <Skeleton variant="text" width="100%" />
                </Box>
              </Box>
            ))}
          </Box>
        )}

        {!isLoading && uploadedCVs.length === 0 && (
          <Typography variant="caption" color="secondary">
            No uploaded CVs found.
          </Typography>
        )}

        {!isLoading &&
          uploadedCVs.map((cv) => {
            const disabled = !isCollectPhase;
            return (
              <Box
                key={cv.upload_id}
                onClick={() => {
                  if (disabled) return;
                  onSelect(cv);
                }}
                sx={{
                  display: "flex",
                  alignItems: "center",
                  columnGap: theme.fixedSpacing(theme.tabiyaSpacing.sm),
                  "&:hover": {
                    backgroundColor: disabled ? "transparent" : theme.palette.action.hover,
                    cursor: disabled ? "default" : "pointer",
                  },
                  opacity: disabled ? 0.5 : 1,
                }}
              >
                <DescriptionOutlinedIcon
                  sx={{ color: theme.palette.text.secondary }}
                  data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_DESCRIPTION_ICON}
                />
                <Box sx={{ display: "flex", flexDirection: "column" }}>
                  <Typography
                    variant="caption"
                    color="secondary"
                    data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_FILE_NAME}
                  >
                    {cv.filename}
                  </Typography>
                  <Typography
                    variant="caption"
                    sx={{ fontSize: theme.fixedSpacing(theme.tabiyaSpacing.sm * 1.3) }}
                    data-testid={DATA_TEST_ID.UPLOADED_CVS_MENU_UPLOAD_DATE}
                  >
                    {getDurationFromNow(new Date(cv.uploaded_at))}
                  </Typography>
                </Box>
              </Box>
            );
          })}
      </Box>
    </Box>
  );
};

export default UploadedCVsMenu;
