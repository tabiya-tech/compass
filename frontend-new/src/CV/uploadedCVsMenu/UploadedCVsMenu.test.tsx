// mute the console
import "src/_test_utilities/consoleMock";

import { render, screen, userEvent } from "src/_test_utilities/test-utils";
import UploadedCVsMenu, { DATA_TEST_ID } from "src/CV/uploadedCVsMenu/UploadedCVsMenu";
import { ConversationPhase } from "src/chat/chatProgressbar/types";
import { CVListItem } from "src/CV/CVService/CVService.types";

// mock help tip component
jest.mock("src/theme/HelpTip/HelpTip", () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation((props) => <div data-testid={props["data-testid"]} />),
  };
});

describe("UploadedCVsMenuContent", () => {
  describe("render tests", () => {
    test("should render the Uploaded CVs Menu Content component correctly", () => {
      // GIVEN a component
      const givenComponent = (
        <UploadedCVsMenu
          uploadedCVs={[
            {
              upload_id: "1",
              filename: "foo_bar.pdf",
              uploaded_at: new Date().toISOString(),
              upload_process_state: "COMPLETED",
              experiences_data: [],
            },
          ]}
          onSelect={jest.fn()}
          onBack={jest.fn()}
          currentPhase={ConversationPhase.COLLECT_EXPERIENCES}
          isLoading={false}
        />
      );

      // WHEN component is rendered
      render(givenComponent);

      // THEN expect the uploaded CVs menu content to be visible
      const uploadedCVsMenuContent = screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_CONTENT);
      expect(uploadedCVsMenuContent).toBeInTheDocument();
      // AND expect the uploaded CVS arrow back icon to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_ARROW_BACK_ICON)).toBeInTheDocument();
      // AND expect the uploaded CVs uploaded text to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_UPLOADED_TEXT)).toBeInTheDocument();
      // AND expect the uploaded CVs help tip to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_HELP_TIP)).toBeInTheDocument();
      // AND expect the uploaded CVs file name to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_FILE_NAME)).toBeInTheDocument();
      // AND expect the uploaded CVs upload date to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_UPLOAD_DATE)).toBeInTheDocument();
      // AND expect the uploaded CVs description icon to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_DESCRIPTION_ICON)).toBeInTheDocument();
      // AND the component to match the snapshot
      expect(uploadedCVsMenuContent).toMatchSnapshot();
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should render correctly when loading", () => {
      // GIVEN a component that is loading
      const givenIsLoading = true;

      // WHEN component is rendered
      render(
        <UploadedCVsMenu
          uploadedCVs={[]}
          onSelect={jest.fn()}
          onBack={jest.fn()}
          currentPhase={ConversationPhase.COLLECT_EXPERIENCES}
          isLoading={givenIsLoading}
        />
      );

      // THEN expect the uploaded CVs menu content to be visible
      const uploadedCVsMenuContent = screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_CONTENT);
      expect(uploadedCVsMenuContent).toBeInTheDocument();
      // AND the skeleton to be visible
      expect(screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_SKELETON)).toBeInTheDocument();
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });
  });

  describe("action tests", () => {
    test("should call onSelect when a CV is clicked", async () => {
      // GIVEN a mock onSelect function
      const mockOnSelect = jest.fn();
      // AND a mock uploaded CV
      const mockUploadedCv: CVListItem = {
        upload_id: "1",
        filename: "foo_bar.pdf",
        uploaded_at: new Date().toISOString(),
        upload_process_state: "COMPLETED",
        experiences_data: ["foo", "bar"],
      };
      // AND the component is rendered
      render(
        <UploadedCVsMenu
          uploadedCVs={[mockUploadedCv]}
          onSelect={mockOnSelect}
          onBack={jest.fn()}
          currentPhase={ConversationPhase.COLLECT_EXPERIENCES}
          isLoading={false}
        />
      );

      // WHEN the user clicks on the uploaded CV item
      const cvItem = screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_FILE_NAME);
      await userEvent.click(cvItem);

      // THEN expect the mock onSelect function to have been called with the correct CV
      expect(mockOnSelect).toHaveBeenCalledWith(mockUploadedCv);
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should call onBack when the back button is clicked", async () => {
      // GIVEN a mock onBack function
      const mockOnBack = jest.fn();
      // AND the component is rendered
      render(
        <UploadedCVsMenu
          uploadedCVs={[]}
          onSelect={jest.fn()}
          onBack={mockOnBack}
          currentPhase={ConversationPhase.COLLECT_EXPERIENCES}
          isLoading={false}
        />
      );

      // WHEN the user clicks on the back button
      const backButton = screen.getByTestId(DATA_TEST_ID.UPLOADED_CVS_MENU_ARROW_BACK_ICON);
      await userEvent.click(backButton);

      // THEN expect the mock onBack function to have been called
      expect(mockOnBack).toHaveBeenCalled();
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });
  });
});
