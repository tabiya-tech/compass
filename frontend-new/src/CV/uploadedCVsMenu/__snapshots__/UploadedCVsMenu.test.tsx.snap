// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadedCVsMenuContent render tests should render the Uploaded CVs Menu Content component correctly 1`] = `
<div
  class="MuiBox-root css-1kkx4gx"
  data-testid="uploaded-cvs-menu-content-e9b40dcc-8068-45ec-a96a-9feb823f1696"
>
  <div
    class="MuiBox-root css-t5n7v9"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-pawek1-MuiButtonBase-root-MuiIconButton-root"
      data-testid="uploaded-cvs-menu-arrow-back-icon-e9b40dcc-8068-45ec-a96a-9feb823f1696"
      tabindex="0"
      title="Back to main menu"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-1rof9ss-MuiTypography-root"
      data-testid="uploaded-cvs-menu-uploaded-text-e9b40dcc-8068-45ec-a96a-9feb823f1696"
    >
      Previously uploaded CVs (1)
    </h6>
    <div
      data-testid="uploaded-cvs-menu-help-tip-e9b40dcc-8068-45ec-a96a-9feb823f1696"
    />
  </div>
  <div
    class="MuiBox-root css-1tkd0ut"
  >
    <div
      class="MuiBox-root css-1unvmzf"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1de4ipc-MuiSvgIcon-root"
        data-testid="uploaded-cvs-menu-description-icon-e9b40dcc-8068-45ec-a96a-9feb823f1696"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm4 18H6V4h7v5h5z"
        />
      </svg>
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1qx6boj-MuiTypography-root"
          data-testid="uploaded-cvs-menu-file-name-e9b40dcc-8068-45ec-a96a-9feb823f1696"
        >
          foo_bar.pdf
        </span>
        <span
          class="MuiTypography-root MuiTypography-caption css-a6pqw0-MuiTypography-root"
          data-testid="uploaded-cvs-menu-upload-date-e9b40dcc-8068-45ec-a96a-9feb823f1696"
        >
          just now
        </span>
      </div>
    </div>
  </div>
</div>
`;
